lib/arjdbc/abstract/core.rb

```diff
@@ -10,39 +10,26 @@ module Core
       def initialize(connection, logger = nil, config = {})
         @config = config
 
+        if self.class.equal? ActiveRecord::ConnectionAdapters::JdbcAdapter
```

Comment:
its quite ugly but with all the mixins the whole `super` setup is quite hard to get right.
\+ JNDI will complicate this a bit more - although that part should be mostly done here as well

since Core was extracted from JdbcAdapter (which used to be the base class 4 all) to be shared and due the order of supers I think this was the only way I got it to work ... the proper configuration for 'dynamic' adapter setups needs to be filled in early - its a legacy feature, but its not yet 'removed'.

it mostly seemed like the core extraction smh forgot about this as I reviewed the prev. changes.
we can remove it but I would do the ugliest thing - comment it with a note that if any adapter extending JdbcAdapter is brought to live for 5.x this needs to be un-commented ... otherwise it might take a few hours of figuring for the next person hitting this :smiling_imp:

Reply:
I'm fine with whatever you think is best, I'm just leery of code that checks to see if it is a specific class before doing something.